File: code/f2237.pas
var i, n, dmax, k: integer;
    a : array[1..1000] of integer;
BEGIN
    readln(n);
    for i:= 1 to n do read(a[i]);
    readln(k);
    dmax:=-1;
    for i:= 1 to n do
        if (a[i]<=k) and (a[i]>dmax) then
            dmax:=a[i];
    write(dmax)
END.

File: code/f2237.cpp
#include <iostream>
using namespace std;
int main()
{
    int i,n, a[1000],k,dmax;
    cin>>n;
    for (i=1; i<=n; i++) cin>>a[i];
    cin>>k;
    dmax=-1;
    for (i=1; i<=n; i++) 
        if (a[i]<=k && a[i]>dmax)
            dmax=a[i];
    cout<<dmax;
    return 0;
}